robotd/cvcapture.py

```diff
@@ -0,0 +1,65 @@
+import threading
+
+from robotd.native import _cvcapture
+
+class CaptureDevice(object):
+    def __init__(self, path=None):
+        if path is not None:
+            argument_c = _cvcapture.ffi.new(
+                'char[]',
+                path.encode('utf-8'),
+            )
+        else:
+            argument_c = _cvcapture.ffi.NULL
+        self.instance = _cvcapture.lib.cvopen(argument_c)
+        self.lock = threading.Lock()
+
+    def capture(self, width, height):
+        if self.instance is None:
+            raise RuntimeError("capture device is closed")
+
+        capture_buffer = _cvcapture.ffi.new(
+            'uint8_t[{}]'.format(width * height),
+        )
+
+        with self.lock:
+            status = _cvcapture.lib.cvcapture(
+                self.instance,
+                capture_buffer,
+                width,
+                height,
+            )
+
+        if status == 0:
+            raise RuntimeError("cvcapture() failed")
+
+        return bytes(_cvcapture.ffi.buffer(capture_buffer))
+
+    def __enter__(self):
```

Comment:
Not sure what the point of implementing context manager protocol here is; nothing seems to use it.